platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,1036 @@
+| TSS-001        | The ledger MUST have a public/private signing key pair that does not change when the consensus roster changes.                        |
+| TSS-002        | The ledger private key MUST be a secret that nobody knows.                                                                            |
+| TSS-003        | The `ledger id` MUST be a public key that everyone knows and that is able to verify signatures from the ledger private key.           |
+| TSS-004        | The ledger signature MUST be produced through aggregating signatures from consensus nodes representing at least 1/3 consensus weight. |
+| TSS-005        | There MUST be a way to bootstrap an existing network's consensus roster with the ledger public/private signing key                    |
+| TSS-006        | There MUST be a way for a new network to generate a public/private signing key for the ledger.                                        |
+| TSS-007        | There MUST be a way to rotate the ledger's public/private signing key.                                                                |
+| TSS-008        | The ledger signature SHOULD be verifiable by an EVM smart contract without excessive gas cost.                                        |
+| TSS-009        | The TSS implementation SHOULD be able to switch elliptic curves.                                                                      |
+| TSS-010        | The TSS algorithm SHOULD be able to model consensus weight with high precision.                                                       |
+| TSS-011        | The TSS design SHOULD minimize the number of steps a Node Operator is required to perform.                                            |
+
+Block Signing Requirements
+
+| Requirement ID | Requirement Description                                                                     |
+|----------------|---------------------------------------------------------------------------------------------|
+| TSSBS-001      | The initial block in the block stream SHOULD be signable and verifiable with the ledger id. |
+| TSSBS-002      | On a new network, the initial block MUST be block 1                                         |
+
+### Design Decisions
+
+Since the consensus node does not depend on TSS capabilities, a new `TSS Base Service` will be created in the
+application to manage state changes and startup behavior. The `TSS Base Service` will be responsible for:
+
+1. Keying the candidate roster with the TSS key material.
+2. On Startup: Determining whether to adopt the candidate roster as the active roster for the platform.
+
+#### New Address Book Life-Cycle
+
+Each day, on the first transaction after midnight, after the staking information has been updated by the app, it will
+generate a new roster and call a public API on `TSS Base Service`, which will then use this as the new candidate
```

Comment:
Each day, we generate a new roster. We will have many new rosters before upgrade. The follow lines assume we only have one new roster before upgrade.

Reply:
It is the latest roster that will be adopted.   I am updating the wording for greater clarity.